### ISSUE TITLE
New rule to prevent awaited renders

### ISSUE THREAD

**Issue by Belco90** [testing-library/eslint-plugin-testing-library#507]
### Name for new rule

no-await-render

### Description of the new rule

Motivated by [this post](https://dev.to/tipsy_dev/testing-library-writing-better-async-tests-c67), there is potential to prevent awaited renders.

### Testing Library feature

All the motivation behind it can be found in the post, but this would relate to the `await` expression + `render` util combined.

### Testing Library framework(s)

All of them with a `render` method: Angular, React, and Vue.

### What category of rule is this?

Warns about a potential error

### Code examples

Invalid example:
```js
it('should render user info', async () => {
  await render(<UserView id="bob" />)
  expect(screen.getByText('Alice and Charlie')).not.toBeNull()
})

```

Valid example:
```js
it('should render user info', async () => {
  render(<UserView id="bob" />)
  expect(await screen.findByText('Alice and Charlie')).not.toBeNull()
})

```

### Do you want to submit a pull request to make the new rule?

Yes
